src/gateways/mappers/beaconResponseMapper.ts

```diff
@@ -0,0 +1,167 @@
+import { IBeacon } from "../../entities/beacon";
+import { IEmergencyContact } from "../../entities/emergencyContact";
+import { IEntityLink } from "../../entities/entityLink";
+import { IOwner } from "../../entities/owner";
+import { IUse } from "../../entities/use";
+import { isoDate } from "../../utils/dateTime";
+import { IBeaconDataAttributes, IBeaconResponse } from "./beaconResponse";
+
+export interface IBeaconResponseMapper {
+  map: (beaconApiResponse: IBeaconResponse) => IBeacon[];
+}
+
+export class BeaconResponseMapper implements IBeaconResponseMapper {
+  public map(beaconApiResponse: IBeaconResponse): IBeacon[] {
+    const result = beaconApiResponse.data.map((b) => {
```

Comment:
You don't need to have the `result` variable here, you can just return the result from the map function.

Also on the inner of the map you don't need to have the `{}` brackets you could do:

```typescript
return beaconApiResponse.data.map((b: IBeaconDataAttributes) => ({
        id: b.id,
        hexId: b.attributes.hexId,
        type: b.attributes.type || "",
        manufacturer: b.attributes.manufacturer || "",
        model: b.attributes.model || "",
        status: b.attributes.status || "",
        registeredDate: isoDate(b.attributes.createdDate || ""),
        batteryExpiryDate: isoDate(b.attributes.batteryExpiryDate || ""),
        chkCode: b.attributes.chkCode || "",
        protocolCode: b.attributes.protocolCode || "",
        codingMethod: b.attributes.codingMethod || "",
        lastServicedDate: isoDate(b.attributes.lastServicedDate || ""),
        manufacturerSerialNumber: b.attributes.manufacturerSerialNumber || "",
        owners: this.mapOwners(beaconApiResponse, b),
        emergencyContacts: this.mapEmergencyContacts(beaconApiResponse, b),
        uses: this.mapUses(beaconApiResponse, b),
        entityLinks: this.mapLinks(b.links),
});
````